src/Avalonia.Controls/Primitives/TemplatedControl.cs

```diff
@@ -326,6 +326,23 @@ protected virtual void OnTemplateChanged(AvaloniaPropertyChangedEventArgs e)
             InvalidateMeasure();
         }
 
+        /// <summary>
+        /// Sets the TemplatedParent property for the created template children.
+        /// </summary>
+        /// <param name="control">The control.</param>
+        private void ApplyTemplatedParent(IControl control)
```

Comment:
What happens when controls are added after ApplyTemplate? What happens if some control is detached? I don't see any code in OnAttached/DetachedToVisualTree that tracks TemplatedParent property of the parent control.